metrics/internal/collectors/storageconsumer.go

```diff
@@ -0,0 +1,123 @@
+		ret = append(ret, m.(*ocsv1alpha1.StorageConsumer))
+	})
+	return ret, err
+}
+
+func NewStorageConsumerCollector(opts *options.Options) *StorageConsumerCollector {
+	ocsClient, err := getocsClient(opts)
+	if err != nil {
+		klog.Error(err)
+	}
+	lw := cache.NewListWatchFromClient(ocsClient, "storageconsumers", metav1.NamespaceAll, fields.Everything())
+	sharedIndexInformer := cache.NewSharedIndexInformer(lw, &ocsv1alpha1.StorageConsumer{}, 0, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	return &StorageConsumerCollector{
+		StorageConsumerMetadata: prometheus.NewDesc(
+			prometheus.BuildFQName("ocs", "storage_consumer", "metadata"),
+			`Attributes of OCS Storage Consumers`,
+			[]string{"storage_consumer_name", "capacity", "state", "granted_capacity"},
+			nil,
+		),
+		Informer: sharedIndexInformer,
+	}
+}
+
+func (c *StorageConsumerCollector) Collect(ch chan<- prometheus.Metric) {
+	storageConsumerLister := NewStorageConsumerLister(c.Informer.GetIndexer())
+	storageConsumers := getAllStorageConsumers(storageConsumerLister, c.AllowedNamespace)
+	if len(storageConsumers) > 0 {
+		c.collectStorageConsumersMetadata(storageConsumers, ch)
+	}
+}
+
+func (c *StorageConsumerCollector) Describe(ch chan<- *prometheus.Desc) {
+	ds := []*prometheus.Desc{
+		c.StorageConsumerMetadata,
+	}
+	for _, d := range ds {
+		ch <- d
+	}
+}
+
+func (c *StorageConsumerCollector) Run(stopCh <-chan struct{}) {
+	go c.Informer.Run(stopCh)
+}
+
+func (c *StorageConsumerCollector) collectStorageConsumersMetadata(storageConsumers []*ocsv1alpha1.StorageConsumer, ch chan<- prometheus.Metric) {
+	for _, storageConsumer := range storageConsumers {
+		ch <- prometheus.MustNewConstMetric(c.StorageConsumerMetadata,
+			prometheus.GaugeValue, 1,
+			storageConsumer.Name,
+			storageConsumer.Spec.Capacity.String(),
+			string(storageConsumer.Status.State),
+			storageConsumer.Status.GrantedCapacity.String())
+	}
+}
+
+func getAllStorageConsumers(lister StorageConsumerLister, namespace string) (storageConsumers []*ocsv1alpha1.StorageConsumer) {
+	storageConsumers, err := lister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("couldn't list StorageConsumer. %v", err)
+	}
+	return
+}
+func getocsClient(opts *options.Options) (*rest.RESTClient, error) {
+	ocsConfig, err := clientcmd.BuildConfigFromFlags("", opts.KubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+	scheme := runtime.NewScheme()
+	utilruntime.Must(ocsv1alpha1.AddToScheme(scheme))
+	codecs := serializer.NewCodecFactory(scheme)
+	ocsConfig.GroupVersion = &ocsv1alpha1.GroupVersion
+	ocsConfig.NegotiatedSerializer = serializer.WithoutConversionCodecFactory{CodecFactory: codecs}
+	ocsConfig.APIPath = "/apis"
+	ocsConfig.ContentType = runtime.ContentTypeJSON
+	if ocsConfig.UserAgent == "" {
+		ocsConfig.UserAgent = rest.DefaultKubernetesUserAgent()
+	}
+	ocsClient, err := rest.RESTClientFor(ocsConfig)
+	if err != nil {
+		return nil, err
+	}
+	return ocsClient, nil
+}
```

Comment:
Can we move this out to a separate file under the same package and export that? At the moment, `codegen` (for generating clientsets) is buggy, so for the time-being it'd make sense to use the same code for client creation. Doing so would help maintain it for the short-term, till this is fixed.

Reply:
@rexagod any suggestions where this can be shifted to?

Reply:
I'd say we can keep this in `collectors/ocs-client.go` (something along those lines), as long as this is exclusive to different collectors.

cc @umangachapagain

Reply:
moved it to `collectors/ocs-client.go`